extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/AbstractJpaOperations.java

```diff
@@ -196,8 +196,11 @@ public PanacheQueryType find(Class<?> entityClass, String query, Object... param
     public PanacheQueryType find(Class<?> entityClass, String query, Sort sort, Object... params) {
         String findQuery = PanacheJpaUtil.createFindQuery(entityClass, query, paramCount(params));
         EntityManager em = getEntityManager(entityClass);
-        // FIXME: check for duplicate ORDER BY clause?
         if (PanacheJpaUtil.isNamedQuery(query)) {
+            if (sort != null) {
+                throw new IllegalArgumentException(
+                        "Sort cannot be used with named queries, add an order by clause to you named query instead");
```

Comment:
Minor typo:
```suggestion
                        "Sort cannot be used with named queries, add an order by clause to your named query instead");
```

But I think we should display the named query used here?

Reply:
right would be nice to call out the named query name

Reply:
Good suggestion! I added it.